pxr/imaging/hdSt/materialNetwork.cpp

```diff
@@ -1121,7 +1121,7 @@ HdStMaterialNetwork::ProcessMaterialNetwork(
 
 #ifdef PXR_MATERIALX_SUPPORT_ENABLED
         if (!isVolume) {
-            _materialXGfx = HdSt_ApplyMaterialXFilter(&surfaceNetwork, materialId,
+            _materialXCodegenResult = HdSt_ApplyMaterialXFilter(&surfaceNetwork, materialId,
```

Comment:
This used to return a shared pointer to a `MaterialX::Shader` object. Unfortunately, a `MaterialX::Shader` is not serializable, and source code is only one part of its state. So, to make caching feasible, I extract all the data that the downstream pipeline needs from `MaterialX::Shader`, encapsulate it in an object called "codegen result" and cache that to disk.